### PR TITLE
Avoid relying on custom endianness Module file

### PR DESCRIPTION
**Pull request by bmanga**
Currently Emscripten relies on a custom TestBigEndian.cmake file that overrides cmake's default one.
This patch proposes to manually specify the endianness in the Platform file, and use the default logic instead.

Otherwise, a problem arises when using vcpkg's emscripten triplet (see [this issue](https://github.com/microsoft/vcpkg/issues/17750))

### CODE REVIEW

**Review by kripken — COMMENTED** [review]
We have an option `SUPPORT_BIG_ENDIAN` - would this change conflict with that?

**Comment by bmanga** [comment]
@kripken thanks for having a look :)
 I don't think so. The custom TestBigEndian.cmake just forced the little endianness, and this PR does the same - the behavior should be identical.
I'm not familiar enough with the option to say whether the existing implementation is correct however. If it's not, I think it should be easy to conditionally set the variable in the platform file rather than adding logic to an overridden module file.

File: cmake/Modules/Platform/Emscripten.cmake [hunk 1]
```diff
@@ -25,6 +25,11 @@ set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS FALSE)
 # OpenCV) use this to detect bitness.
 set(CMAKE_SYSTEM_PROCESSOR x86)
 
+# Since emscripten produces Javascript, we set the byte order to little endian
+# to avoid cmake's checks.
```
Comment (sbc100) [inline cmake/Modules/Platform/Emscripten.cmake:29]: How about just "WebAssembly is always little endian so avoid the cmake's checks and force this."

**Comment by sbc100** [comment]
This change seems good to me.   If we want to support bigendian + cmake (we don't today) we could make just delete TestBigEndian.cmake and let the checks run?

For now, I think this is good to land.

**Comment by bmanga** [comment]
@sbc100 thanks for the review :D
CMake's checks currently fail to determine the endianness for emscripten, so we either have the option of manually specifying it with the variables (like done here) or sending a PR to cmake.

**Comment by kripken** [comment]
It looks like the CI failure on `other.test_cmake_js` might be real, as it sounds related to cmake?

**Comment by bmanga** [comment]
@kripken Is there a way to check the logs of the failure?

Taking a look at the test, it could be a failure from this part of the CMakeLists.txt:
```
include(TestBigEndian)
test_big_endian(endian_result)
if (NOT ${endian_result} EQUAL 0)
        message(FATAL_ERROR "TEST_BIG_ENDIAN did not return 0!")
endif()
```
But the `Emscripten.cmake` toolchain file should be loaded by that point, correct?

**Comment by bmanga** [comment]
Well I think I found the problem. Those CMake variables are only supported starting from version 3.20 :/

**Comment by kripken** [comment]
Oh, interesting @bmanga ... if they require a pretty new cmake then I think we can't depend on them. We don't specify a specific version of cmake that I can see, but I don't think we want to depend on a recent one.

**Pull request closed without merging**